core/tee/fs_htree.c

```diff
@@ -473,8 +474,13 @@ static TEE_Result authenc_init(void **ctx_ret, TEE_OperationMode mode,
 		goto err_free;
 
 	if (!ni) {
+		size_t hash_size = TEE_FS_HTREE_HASH_SIZE;
+
+		if (IS_ENABLED(CFG_REE_FS_HTREE_HASH_SIZE_COMPAT))
+			hash_size = TEE_FS_HTREE_FEK_SIZE;
```

Comment:
Please fix `aad_len` too.

Reply:
`aad_len` already considers `TEE_FS_HTREE_HASH_SIZE` at line 457, even if we used to pass a truncated hash size. I think we should not change (fix) the AAD size in the legacy config (`CFG_REE_FS_HTREE_HASH_SIZE_COMPAT=y`).

Reply:
AES-GCM doesn't code the supplied `aad_len` into the tag or anything. It only accumulates the AAD, but as @anilreddy3 discovered, it might be checked in the AES-GCM implementation. That's why I think that `aad_len` should match the length of the supplied AAD.
If someone (for some reason) replaced AES-GCM with AES-CCM, then `aad_len` must match the length of the supplied AAD.

Reply:
Yes, correct.

Reply:
Ok, thanks @jenswi-linaro for the explanation. @anilreddy3, could you add a dedicated commit for this change, not mix within the commit dedicated to `CFG_REE_FS_HTREE_HASH_SIZE_COMPAT`.

Reply:
I think, if someone replaces AES-GCM with AES-CCM it anyway break the existing file system, In that case they should move to fix version by disabling CFG_REE_FS_HTREE_HASH_SIZE_COMPAT config, with that aad_len match with the supplied AAD data.